cloud-minecraft/cloud-fabric/src/main/java/cloud/commandframework/fabric/FabricClientCommandManager.java

```diff
@@ -0,0 +1,114 @@
+            final Function<CommandTree<FabricClientCommandSource>, CommandExecutionCoordinator<FabricClientCommandSource>> execCoordinator
+    ) {
+        return new FabricClientCommandManager<>(execCoordinator, Function.identity(), Function.identity());
+    }
+
+    /**
+     * Create a new command manager instance.
+     *
+     * @param commandExecutionCoordinator  Execution coordinator instance. The coordinator is in charge of executing incoming
+     *                                     commands. Some considerations must be made when picking a suitable execution coordinator
+     *                                     for your platform. For example, an entirely asynchronous coordinator is not suitable
+     *                                     when the parsers used in that particular platform are not thread safe. If you have
+     *                                     commands that perform blocking operations, however, it might not be a good idea to
+     *                                     use a synchronous execution coordinator. In most cases you will want to pick between
+     *                                     {@link CommandExecutionCoordinator#simpleCoordinator()} and
+     *                                     {@link AsynchronousCommandExecutionCoordinator}
+     * @param commandSourceMapper          Function that maps {@link FabricClientCommandSource} to the command sender type
+     * @param backwardsCommandSourceMapper Function that maps the command sender type to {@link FabricClientCommandSource}
+     */
+    @SuppressWarnings("unchecked")
+    public FabricClientCommandManager(
+            final @NonNull Function<@NonNull CommandTree<C>, @NonNull CommandExecutionCoordinator<C>> commandExecutionCoordinator,
+            final Function<FabricClientCommandSource, C> commandSourceMapper,
+            final Function<C, FabricClientCommandSource> backwardsCommandSourceMapper
+    ) {
+        super(
+                commandExecutionCoordinator,
+                commandSourceMapper,
+                backwardsCommandSourceMapper,
+                new FabricCommandRegistrationHandler.Client<>(),
+                () -> (FabricClientCommandSource) new ClientCommandSource(
+                        MinecraftClient.getInstance().getNetworkHandler(),
+                        MinecraftClient.getInstance()
+                )
+        );
+
+        this.registerParsers();
+    }
+
+    private void registerParsers() {
+    }
+
+    /**
+     * Check if a sender has a certain permission.
+     *
+     * <p>The implementation for client commands always returns true.</p>
+     *
+     * @param sender     Command sender
+     * @param permission Permission node
+     * @return whether the sender has the specified permission
+     */
+    @Override
+    public boolean hasPermission(@NonNull final C sender, @NonNull final String permission) {
+        return true;
```

Comment:
We should probably provide a way to override this, or perhaps some default predicate permissions?

Some things that might want to be checked:

- whether or not cheats are enabled
- the server instructing a certain mod to not function via some mod-determined protocol

Both of these are probably best done with predicate permissions though...

Reply:
I've added some predicates for whether or not cheats are enabled - not sure about the mod-determined protocol stuff, maybe it's best to leave to mod authors?